alchemy/src/vercel/project.ts

```diff
@@ -0,0 +1,418 @@
+
+  /**
+   * Specifies whether Zero Config Failover is enabled for this project.
+   */
+  serverlessFunctionZeroConfigFailover?: boolean;
+
+  /**
+   * OpenID Connect JSON Web Token generation configuration.
+   */
+  oidcTokenConfig?: {
+    /**
+     * Whether or not to generate OpenID Connect JSON Web Tokens.
+     */
+    enabled: boolean;
+
+    /**
+     * team: `https://oidc.vercel.com/[team_slug]` global: `https://oidc.vercel.com`
+     */
+    issuerMode: "team" | "global";
+  };
+
+  /**
+   * Opt-in to skip deployments when there are no changes to the root directory and its dependencies
+   */
+  enableAffectedProjectsDeployments?: boolean;
+
+  /**
+   * Resource configuration
+   */
+  resourceConfig?: {
+    /**
+     * Whether fluid is enabled
+     */
+    fluid?: boolean;
+
+    /**
+     * Default regions for functions
+     */
+    functionDefaultRegions?: string[];
+
+    /**
+     * Default timeout for functions
+     */
+    functionDefaultTimeout?: number;
+
+    /**
+     * Default memory type for functions
+     */
+    functionDefaultMemoryType?: "standard_legacy" | "standard" | "performance";
+
+    /**
+     * Whether function zero config failover is enabled
+     */
+    functionZeroConfigFailover?: boolean;
+
+    /**
+     * Whether elastic concurrency is enabled
+     */
+    elasticConcurrencyEnabled?: boolean;
+
+    /**
+     * The build machine type
+     */
+    buildMachineType?: "enhanced" | "ultra";
+  };
+}
+
+/**
+ * Output returned after Project creation/update
+ */
+export interface Project extends Resource<"vercel::Project">, ProjectProps {
+  /**
+   * The ID of the project
+   */
+  id: string;
+
+  /**
+   * The account ID that the project belongs to
+   */
+  accountId: string;
+
+  /**
+   * The time at which the project was created
+   */
+  createdAt: number;
+
+  /**
+   * The time at which the project was last updated
+   */
+  updatedAt: number;
+
+  /**
+   * The latest deployment of the project
+   */
+  latestDeployment?: {
+    /**
+     * The ID of the deployment
+     */
+    id: string;
+
+    /**
+     * The URL of the deployment
+     */
+    url: string;
+  };
+}
+
+export const Project = Resource(
+  "vercel::Project",
+  async function (
+    this: Context<Project>,
+    id: string,
+    { accessToken, ...props }: ProjectProps & { accessToken?: Secret },
+  ): Promise<Project> {
+    switch (this.phase) {
+      case "delete": {
+        const api = await createVercelApi({
+          baseUrl: "https://api.vercel.com/v9",
+          accessToken,
+        });
+
+        try {
+          if (this.output?.id) {
+            const deleteResponse = await api.delete(
+              `/projects/${this.output.id}`,
+            );
+
+            if (!deleteResponse.ok && deleteResponse.status !== 404) {
+              console.error(
+                "Error deleting project:",
+                deleteResponse.statusText,
+              );
+            }
```

Comment:
for each `create`, `update`, `delete` api call, pul it out and export from the `project.ts` module

```ts
export function deleteProject(api: VercelApi, projectId: string) {
  const deleteResponse = await api.delete(
    `/projects/${projectId}`,
  );

  if (!deleteResponse.ok && deleteResponse.status !== 404) {
    console.error(
      "Error deleting project:",
      deleteResponse.statusText,
    );
  }
}
```

Reply:
Abstracted out `createProject` and `updateEnvironmentVariables`. 

Turns out `deleteProject` would've been 1 line (`await api.delete`)`, which doesn't seem worth it...

Resolved via dbb983752908bb9737e002c4296e8e223cf8c26a